### PR TITLE
Fix parameter restorage after loading EMA parameters 

### PR DESCRIPTION
**Pull request by KiddoZhu**
This PR fixes a serious bug in training. The original code loads EMA parameters during validation, but fails to restore the training parameters after validation.

In [these lines](https://github.com/aqlaboratory/openfold/blob/main/train_openfold.py#L127-L129), `self.cached_weights = self.model.state_dict()` only keeps a reference to the parameters, rather than a copy. However, `self.model.load_state_dict()` performs an in-place update to the parameters, which unintentionally changes the parameters in `self.cached_weights`. Therefore, [the callback](https://github.com/aqlaboratory/openfold/blob/main/train_openfold.py#L143-L146) at the end of validation step fails to restore the correct parameters.

In other words, the current implementation always replaces the training parameters with EMA parameters at the end of each epoch. Empirically, I found this slows down the convergence very much.

### CODE REVIEW

**Comment by gahdritz** [comment]
You're right. Thanks for bringing this to our attention.
